mobile/src/hooks/use-feed.ts

```diff
@@ -4,11 +4,11 @@ import { useGno } from "./use-gno";
 export const useFeed = () => {
   const gno = useGno();
 
-  async function fetchFeed(startIndex: number, endIndex: number): Promise<{ data: Post[]; n_threads: number }> {
+  async function fetchFeed(startIndex: number, endIndex: number): Promise<{ data: Post[]; n_posts: number }> {
     const currentAccount = await gno.getActiveAccount();
     if (!currentAccount.key) throw new Error("No active account");
     const bech32 = await gno.addressToBech32(currentAccount.key.address);
-    const result = await gno.qEval("gno.land/r/berty/social", `GetThreadPosts("${bech32}" , 0, 0, ${startIndex}, ${endIndex})`);
+    const result = await gno.qEval("gno.land/r/berty/social", `GetJsonHomePosts("${bech32}", ${startIndex}, ${endIndex})`);
     const json = convertToPosts(result, currentAccount.key.name);
```

Comment:
Right now, `convertToPosts` always shows the current account name as the poster. But a followed post has a different user. Need to look at the `creator` field of the result from `GetJsonHomePosts`.

Reply:
The web page Render calls displayAddressMD which calls `GetUserByAddress` every time to get the post's user name.
https://github.com/gnolang/gnosocial/blob/179c149570fd5b6b1c8d75ae392eac26f74862f0/realm/util.gno#L63

Within the Gno node, this is efficient. But calling `GetUserByAddress` remotely from the mobile app is not as efficient. Maybe you need a local cache which maps a user's address to name?

Reply:
Can we have the creator's name in the response as well to simplify and speed up?

Reply:
I think we need to discuss this. As I mentioned above, right now we have the username associated with the address. But in the future, there will be the display name, user icon and other statistics. If we put all of that in the response, then `GetJsonHomePosts` needs to compute it, and we need to use the connection bandwidth to transmit all of it (repeated again and again in each post of the reply).